models/reading.go

```diff
@@ -20,31 +20,37 @@ import (
-	Pushed      int64  `json:"pushed,omitempty" codec:"pushed,omitempty"`   // When the data was pushed out of EdgeX (0 - not pushed yet)
-	Created     int64  `json:"created,omitempty" codec:"created,omitempty"` // When the reading was created
-	Origin      int64  `json:"origin,omitempty" codec:"origin,omitempty"`
-	Modified    int64  `json:"modified,omitempty" codec:"modified,omitempty"`
-	Device      string `json:"device,omitempty" codec:"device,omitempty"`
-	Name        string `json:"name,omitempty" codec:"name,omitempty"`
-	Value       string `json:"value,omitempty"  codec:"value,omitempty"`            // Device sensor data value
-	BinaryValue []byte `json:"binaryValue,omitempty" codec:"binaryValue,omitempty"` // Binary data payload
-	isValidated bool   // internal member used for validation check
+	Id            string `json:"id,omitempty" codec:"id,omitempty"`
+	Pushed        int64  `json:"pushed,omitempty" codec:"pushed,omitempty"`   // When the data was pushed out of EdgeX (0 - not pushed yet)
+	Created       int64  `json:"created,omitempty" codec:"created,omitempty"` // When the reading was created
+	Origin        int64  `json:"origin,omitempty" codec:"origin,omitempty"`
+	Modified      int64  `json:"modified,omitempty" codec:"modified,omitempty"`
+	Device        string `json:"device,omitempty" codec:"device,omitempty"`
+	Name          string `json:"name,omitempty" codec:"name,omitempty"`
+	Value         string `json:"value,omitempty" codec:"value,omitempty"` // Device sensor data value
+	ValueType     string `json:"valueType,omitempty" codec:"valueType,omitempty"`
+	FloatEncoding string `json:"floatEncoding,omitempty" codec:"floatEncoding,omitempty"`
+	BinaryValue   []byte `json:"binaryValue,omitempty" codec:"binaryValue,omitempty"` // Binary data payload
+	MediaType     string `json:"mediaType,omitempty" codec:"mediaType,omitempty"`
+	isValidated   bool   // internal member used for validation check
 }
 
 // UnmarshalJSON implements the Unmarshaler interface for the Reading type
 func (r *Reading) UnmarshalJSON(data []byte) error {
 	var err error
 	type Alias struct {
-		Id          *string `json:"id"`
-		Pushed      int64   `json:"pushed"`
-		Created     int64   `json:"created"`
-		Origin      int64   `json:"origin"`
-		Modified    int64   `json:"modified"`
-		Device      *string `json:"device"`
-		Name        *string `json:"name"`
-		Value       *string `json:"value"`
-		BinaryValue []byte  `json:"binaryValue"`
+		Id            *string `json:"id"`
+		Pushed        int64   `json:"pushed"`
+		Created       int64   `json:"created"`
+		Origin        int64   `json:"origin"`
+		Modified      int64   `json:"modified"`
+		Device        *string `json:"device"`
+		Name          *string `json:"name"`
+		Value         *string `json:"value"`
+		ValueType     *string `json:"valueType"`
+		FloatEncoding *string `json:"floatEncoding"`
+		BinaryValue   []byte  `json:"binaryValue"`
+		MediaType     *string `json:"mediaType"`
```

Comment:
As per Trevor's comment: https://github.com/edgexfoundry/go-mod-core-contracts/issues/193#issuecomment-593481163

Will we need to add some additional logic to the `Validate` method to ensure the data is correct?

Reply:
I think you’re referring to the media type only being filled in for binary data? I think that’s not something we’re looking to enforce necessarily, but I’ll look into it and get back to you.

Reply:
Per Anthony's comment, it's a good idea to add logic to `Validate()` so that 
- if `ValueType == Binary` then `MediaType` should not be empty.
- if `ValueType == Float` then `FloatEncoding` should not be empty.

Reply:
Should we also enforce that only the proper "content" is populated? For example,

- if ValueType == Binary then FloatEncoding should be empty.
- if ValueType == Float then MediaType should be empty.

This would protect the user from providing incorrect data, and also protect us from wasting storage space on useless data.

Reply:
@tsconn23 How should I do the comparisons on the ValueType property? Should I define string constants in this file for clients of core-contracts to use to fill it in?  Something like 
```
// Constants related to Reading ValueTypes
const (
	ValueTypeBool    = "Bool"
	ValueTypeString  = "String"
	ValueTypeUint8   = "Uint8"
	ValueTypeUint16  = "Uint16"
	ValueTypeUint32  = "Uint32"
	ValueTypeUint64  = "Uint64"
	ValueTypeInt8    = "Int8"
	ValueTypeInt16   = "Int16"
	ValueTypeInt32   = "Int32"
	ValueTypeInt64   = "Int64"
	ValueTypeFloat32 = "Float32"
	ValueTypeFloat64 = "Float64"
	ValueTypeBinary  = "Binary"
)
```
?

Reply:
@tsconn23 Hm, alright.  I only ask because it would seem we need some way to check the values inside ValueType if we're going to perform the checks such as `if ValueType == Float then FloatEncoding should not be empty.`

For the binary one I suppose we can just check to see if the BinaryData value is non-empty, but I don't see any way around inspecting the value for the Float case.  Unless someone has a better idea I suppose I'll create the constants I suggested above.

The enum referenced above is an integer enum--won't be useful here.

Reply:
@AnthonyMBonafide @tsconn23 Implemented these checks and added test cases.  Please take a look and give any feedback you may have!